pandas/tests/extension/test_arrow.py

```diff
@@ -1739,6 +1739,21 @@ def test_setitem_invalid_dtype(data):
         data[:] = fill_value
 
 
+@pytest.mark.skipif(pa_version_under8p0, reason="returns object with 7.0")
+def test_from_arrow_respecting_given_dtype():
```

Comment:
Could you add a test for an invalid/unsafe cast as well?

Reply:
added a test